### ISSUE TITLE
"Like" action is not working

### ISSUE THREAD

**Issue by sanadhyp** [alexal1/Insomniac#367]
I have configured the bot to run in "Interact" mode and just like the posts for hashtags. From past few day, may be after the update in Insta and Samsung's android, the liking has stopped working. If you see the logs, it searches for hashtag, scrolls down number of times and then fails to find a post.

`
[07/28 10:36:22] Handle #hotwheelspic
[07/28 10:36:22] There is no special interaction-instructions for #hotwheelspic. Working with #hotwheelspic recent-likers.
[07/28 10:36:22] Interacting with #hotwheelspic-recent-likers
[07/28 10:36:22] Closing keyboard...
[07/28 10:36:23] The device keyboard is already closed.
[07/28 10:36:30] #hotwheelspic is not in recent searching history...
[07/28 10:36:32] Switching to Recent tab
[07/28 10:36:40] Sleep for 5.73 seconds
[07/28 10:36:46] Scroll down 5 times.
[07/28 10:36:47] Sleep for 1.16 seconds
[07/28 10:36:50] Sleep for 1.65 seconds
[07/28 10:36:53] Sleep for 1.92 seconds
[07/28 10:36:57] Sleep for 1.37 seconds
[07/28 10:37:00] Sleep for 1.63 seconds
[07/28 10:37:02] Choosing a random post from those on the screen
[07/28 10:37:02] No posts here. Abort.

[07/28 10:37:02] Handle #bburago
[07/28 10:37:02] There is no special interaction-instructions for #bburago. Working with #bburago recent-likers.
[07/28 10:37:02] Interacting with #bburago-recent-likers
[07/28 10:37:02] Closing keyboard...
[07/28 10:37:03] The device keyboard is already closed.
[07/28 10:37:09] Sleep for 2.70 seconds
[07/28 10:37:12] Switching to Recent tab
[07/28 10:37:14] Sleep for 2.37 seconds
[07/28 10:37:16] Scroll down 4 times.
[07/28 10:37:17] Sleep for 1.44 seconds
[07/28 10:37:21] Sleep for 1.12 seconds
[07/28 10:37:23] Sleep for 1.18 seconds
[07/28 10:37:26] Sleep for 2.21 seconds
[07/28 10:37:29] Choosing a random post from those on the screen
[07/28 10:37:29] No posts here. Abort.
`

I tried to debug, and I believe this piece of code from Views.py is not working as expected:

`        for post_view in posts_grid.child(resourceId=f'{self.device.app_id}:id/image_button',
                                          className='android.widget.ImageView'):
`

nothing is being found from the posts_grid.
[sample.log](https://github.com/alexal1/Insomniac/files/6890486/sample.log)

**Comment by Baghodlr** [comment]
Having this same issue; believe it happened after the most recent IG update.

**Comment by sanadhyp** [comment]
> Having this same issue; believe it happened after the most recent IG update.

Yeah, I believe so. They have changed the posts_grid UI widget it seems.

**Comment by Baghodlr** [comment]
> > Having this same issue; believe it happened after the most recent IG update.
> 
> Yeah, I believe so. They have changed the posts_grid UI widget it seems.

So I'm not sure if it will be of any use to you, but I was having another issue with the bot not finding frames to reference (e.g. TAGS, ACCOUNT, etc.) and downgrading from 3.7.21 to 3.7.20 fixed these problems for me. I am running an emulator using LDPlayer for reference.

**Comment by sanadhyp** [comment]
> > > Having this same issue; believe it happened after the most recent IG update.
> > 
> > 
> > Yeah, I believe so. They have changed the posts_grid UI widget it seems.
> 
> So I'm not sure if it will be of any use to you, but I was having another issue with the bot not finding frames to reference (e.g. TAGS, ACCOUNT, etc.) and downgrading from 3.7.21 to 3.7.20 fixed these problems for me. I am running an emulator using LDPlayer for reference.

Oh I see. I will compare the line of code in question with lower version. Though I doubt it will fix if Instagram has changed something on it's UI. Thanks for sharing.

**Comment by sanadhyp** [comment]
Tagging @alexal1

**Comment by Alex01991** [comment]
@alexal1

**Comment by Alex01991** [comment]
@alexal1 - don't work like and follow

**Comment by Baghodlr** [comment]
Downgrade from app version 200 (or whatever the newest IG versioning is) to 196.X.X.X and it will fix this issue as well as others.
